.github/workflows/ci.yml

```
@@ -108,10 +108,32 @@ jobs:
 
       - name: "Install Chrome"
         uses: browser-actions/setup-chrome@db1b524c26f20a8d1a10f7fc385c92387e2d0477 # v1.7.1
+        id: setup-chrome
+        if: ${{ matrix.nox-session == 'emscripten' }}
+        with:
+          chrome-version: 120
+          install-chromedriver: true
+      - name: Force override system chrome
+        run: |
+          sudo rm -f /usr/bin/google-chrome
+          sudo rm -f /usr/bin/chrome
+          sudo ln -s ${{ steps.setup-chrome.outputs.chrome-path }} /usr/bin/google-chrome
+          sudo ln -s ${{ steps.setup-chrome.outputs.chrome-path }} /usr/bin/chrome
+          google-chrome --version
         if: ${{ matrix.nox-session == 'emscripten' }}
       - name: "Install Firefox"
         uses: browser-actions/setup-firefox@233224b712fc07910ded8c15fb95a555c86da76f # v1.5.0
         if: ${{ matrix.nox-session == 'emscripten' }}
+      - name: "Install node.js"
+        uses: actions/setup-node@v4
+        if: ${{ matrix.nox-session == 'emscripten' }}
+        with:
+          node-version: 22
+      - name: Cache pyodide downloads in nox cache
+        uses: actions/cache@v4
+        with:
+          path: .nox/.cache
+          key: pyodide_downloads
```

I believe the key has to include a hash of `noxfile.py` which contains a Pyodide version.
If the key never changes, the following can happen: when Pyodide version is changed, files of the new version won't be added to the cache until the cache is evicted. And the eviction may never happen if the cache is restored frequently enough.

```suggestion
          # noxfile.py contains a Pyodide version used currently.
          key: pyodide-downloads-${{ hashFiles('noxfile.py') }}
```

---

Also, this step should be limited to the `emscripten` session